contrib/scalar/scalar.c

```diff
@@ -1053,13 +1098,17 @@ static int cmd_diagnose(int argc, const char **argv)
 	fwrite(buf.buf, buf.len, 1, stdout);
 
 	dir_file_stats(&packs_buf, ".git/objects/pack");
+	loose_objs_stats(&objs_buf, ".git/objects");
 
 	if ((res = stage(tmp_dir.buf, &buf, "diagnostics.log")))
 		goto diagnose_cleanup;
 
 	if ((res = stage(tmp_dir.buf, &packs_buf, "packs-local.txt")))
 		goto diagnose_cleanup;
 
+	if ((res = stage(tmp_dir.buf, &objs_buf, "objects-local.txt")))
+		goto diagnose_cleanup;
+
 	if ((res = stage_directory(tmp_dir.buf, ".git", 0)) ||
 	    (res = stage_directory(tmp_dir.buf, ".git/hooks", 0)) ||
 	    (res = stage_directory(tmp_dir.buf, ".git/info", 0)) ||
```

Comment:
Oy, I think I will retract my comment about "best effort": the surrounding code assumes too hard that it should fail early. I do not think we will have to spend time on turning it into best effort, either.